pkg/restore/client.go

```diff
@@ -355,11 +355,24 @@ func (rc *Client) CreateTables(
 	return rewriteRules, newTables, nil
 }
 
-func (rc *Client) createTable(dom *domain.Domain, table *utils.Table, newTS uint64) (CreatedTable, error) {
+func (rc *Client) createTable(
+	ctx context.Context,
+	db *DB,
+	dom *domain.Domain,
+	table *utils.Table,
+	newTS uint64,
+) (CreatedTable, error) {
+	if db == nil {
+		db = rc.db
+	}
```

Comment:
can we remove `rc.db`, or make the DB pool `[]*DB` a field of `rc *Client`?

Reply:
`rc.db` still be used for other DDLs (e.g. `CREATE DATABASE`, `ALTER TABLE SET TIFLASH_REPLICA = x`, incremental DDL execution, etc.), if we want to remove it, we must make many changes. (BTW It's seems a little strange if these uses db like `rc.dbs[n]` outside `GoCreateTables`, there is an array but we just need one of them at anytime, since those DDLs  are executed sequential.).

Maybe do that at another refactoring PR would be better?

Reply:
we could have a `rc.GetDB()` and rotate the db list on every call 🙃.

anyway, as we decide to refactor in the future, please add a TODO on the `db` field.

Reply:
But how `GetDB()` should be implemented? If we trivially return `rc.dbs[0]`, seems we cannot take any good from it... Because we still need to inject some 'id' to every call if we need to use other dbs (hence we can't remove the extra parameter of `createTable`), and we use just one db at once (hence we can't speed up execution outside `restoreTables`).

I think maybe we need farther discussion to make sure whether we need to prune the `db` field... :thinking:

Reply:
@YuJuncen round-robin?

```go
db := rc.dbs[rc.curDB]
rc.curDB++
if rc.curDB >= len(rc.dbs) {
    rc.curDB = 0
}
return db
```

Reply:
Yes, that's a good idea, but seem here would still be some risk if we want to use this algorithm to allocate session for processes concurrently running, that is, when some session stuck:

Assume we have two sessions, and two processes.

```
(Process 0) |(takes dbs[0]) -------------------------------------------------------------------->(done)|
(Process 1)      |(takes dbs[1])----->(done)|    |(takes dbs[0], bomb, race condition!)*
```

Maybe this cannot provide more for us... Our questions still present: we cannot remove the extra parameter of `createTable`, and out of `createTable`, we cannot take good of multi-dbs too, since they are executed sequentially.